src/main/java/com/commercetools/sync/commons/utils/ClientConfigurationUtils.java

```diff
@@ -1,48 +1,22 @@
 package com.commercetools.sync.commons.utils;
 
 import io.sphere.sdk.client.BlockingSphereClient;
-import io.sphere.sdk.client.QueueSphereClientDecorator;
-import io.sphere.sdk.client.RetrySphereClientDecorator;
-import io.sphere.sdk.client.SphereAccessTokenSupplier;
 import io.sphere.sdk.client.SphereClient;
 import io.sphere.sdk.client.SphereClientConfig;
-import io.sphere.sdk.http.AsyncHttpClientAdapter;
-import io.sphere.sdk.http.HttpClient;
-import io.sphere.sdk.retry.RetryAction;
-import io.sphere.sdk.retry.RetryContext;
-import io.sphere.sdk.retry.RetryPredicate;
-import io.sphere.sdk.retry.RetryRule;
-import org.asynchttpclient.AsyncHttpClient;
-import org.asynchttpclient.DefaultAsyncHttpClient;
-import org.asynchttpclient.DefaultAsyncHttpClientConfig;
 
 import javax.annotation.Nonnull;
-import java.time.Duration;
-import java.util.Collections;
-import java.util.List;
 import java.util.concurrent.TimeUnit;
-import java.util.function.Function;
-
-import static io.sphere.sdk.http.HttpStatusCode.BAD_GATEWAY_502;
-import static io.sphere.sdk.http.HttpStatusCode.GATEWAY_TIMEOUT_504;
-import static io.sphere.sdk.http.HttpStatusCode.SERVICE_UNAVAILABLE_503;
 
 public final class ClientConfigurationUtils {
```

Comment:
This class seems redundant now. What's the reason to move all the logic to RetryableSphereClientWithExponentialBackoff?

Reply:
Not redundant, should be changed with care, when you remove it, it's a breaking change for the library user. 
So for now we could keep it, or document a breaking change. 

 https://github.com/commercetools/commercetools-sync-java/blob/master/docs/usage/IMPORTANT_USAGE_TIPS.md#customized-sphereclient-creation

Reply:
The question is why there's an extra class `RetryableSphereClientWithExponentialBackoff`? Because of that class now `ClientConfigurationUtils` is useless. It just call `ClientConfigurationUtils`. Why not leaving all the logic inside `ClientConfigurationUtils`?

Reply:
The RetryableSphereClientWithExponentialBackoff will be moved to SDK, otherwise, you don't need all changes, the idea of this PR is the first review before the SDK team, then move the part to the SDK.